app/services/pages/column.rb

```diff
@@ -0,0 +1,18 @@
+module Pages
+  class Column
+
+    attr_reader :widgets
+
+    def initialize
+      @widgets = []
+    end
+
+    def find_widget(widget_name)
+      @widgets.select { |w| w.name == widget_name.to_sym }.first
+    end
+
+    def <<(widget)
+      @widgets << widget
+    end
+  end
+end
```

Comment:
I think that layout is basically determined by the existing page/tab you are trying to extend. 
If you allow to specify the position by `col-x-y`, you have to be prepared to handle conflicts and I tried to keep it as simple as possible for the beginning, but it can potentially be enhanced into some sort of grid system.